coordinator/src/services/translation/request/return/return.ts

```diff
@@ -1,57 +1,53 @@
 import {hl7V3, fhir, processingErrors as errors} from "@models"
 import {getCodeableConceptCodingForSystem, getMessageId} from "../../common"
 import {convertIsoDateTimeStringToHl7V3DateTime} from "../../common/dateTime"
-import * as pino from "pino"
 import {getMessageIdFromTaskFocusIdentifier, getPrescriptionShortFormIdFromTaskGroupIdentifier} from "../task"
-import Hapi from "@hapi/hapi"
-import {getContainedPractitionerRole} from "../../common/getResourcesOfType"
-import {createAgentPersonFromAuthenticatedUserDetailsAndPractitionerRole} from "../agent-unattended"
+import {getContainedOrganizationViaReference, getContainedPractitionerRole} from "../../common/getResourcesOfType"
+import {createAuthorUsingPractitionerRoleAndOrganization} from "../agent-unattended"
 import {isReference} from "../../../../utils/type-guards"
 
-export async function convertTaskToDispenseProposalReturn(
+export function convertTaskToDispenseProposalReturn(
   task: fhir.Task,
-  headers: Hapi.Util.Dictionary<string>,
-  logger: pino.Logger
-): Promise<hl7V3.DispenseProposalReturn> {
+): hl7V3.DispenseProposalReturn {
   const idValue = getMessageId(task.identifier, "Task.identifier")
   const id = new hl7V3.GlobalIdentifier(idValue)
   const effectiveTime = convertIsoDateTimeStringToHl7V3DateTime(task.authoredOn, "Task.authoredOn")
   const dispenseProposalReturn = new hl7V3.DispenseProposalReturn(id, effectiveTime)
 
-  if (isReference(task.requester)) {
-    const taskPractitionerRole: fhir.PractitionerRole = getContainedPractitionerRole(
-      task,
-      task.requester.reference
-    )
-    dispenseProposalReturn.author = await createAuthor(taskPractitionerRole, headers, logger)
-  } else {
+  if (!isReference(task.requester)) {
     throw new errors.InvalidValueError(
       "For return messages, task.requester must be a reference to a contained PractitionerRole resource."
     )
   }
 
+  const practitionerRole: fhir.PractitionerRole = getContainedPractitionerRole(
+    task,
+    task.requester.reference
+  )
+
+  if (!isReference(practitionerRole.organization)) {
+    throw new errors.InvalidValueError(
+      "For return messages, practitionerRole.organization must be a reference to a contained Organization resource."
```

Comment:
i know this is copied from line 19 but this code will run for both `return` and `withdraw` messages